dist/tools/openocd/openocd.sh

```diff
@@ -202,7 +202,7 @@ do_debug() {
             -c 'halt' \
             -l /dev/null" &
     # save PID for terminating the server afterwards
-    OCD_PID=$?
+    OCD_PID=$(($!+2))
```

Comment:
an idea:
use the `$$` variable in combination with some variation of `sh -c "echo \$\$; xxx" &` and some output redirection

Test with: `setsid sh -c "echo child: \$\$; sleep 2" &; echo "setsid: $\!"; echo "shell: $$"`